dig/README.md

```diff
@@ -0,0 +1,78 @@
+# Dependency Injection Graph
+
+Package dig provides a fairly opinionated way of resolving object dependencies.
+There are two sides of dig: `Inject` and `Resolve`.
```

Comment:
Inject and resolve are of course colloquially "dig" and "dug".

![digdug](https://cloud.githubusercontent.com/assets/60294/23088970/0d3209fc-f537-11e6-97dc-ec0f6f9f8680.jpg)